<?xml version="1.0"?>


<!-- Launches everything except for the actor controller. -->

<launch>

  <arg name="robot_name" default="$(env ROBOT_NAME)"/>
  <arg name="actor_name" default="actor1"/>

  <include file="$(find ariac_environment)/launch/ariac_p3dx.launch"/>

  <!-- Get fake localization data as a transform -->
    <node pkg="fake_localization" type="fake_localization" name="fakeloc">
      <param name="odom_frame_id" value="$(arg robot_name)/odom"/>
      <param name="global_frame_id" value="$(arg robot_name)/map"/>
      <param name="base_frame_id" value="$(arg robot_name)/base_link"/>
      <remap from="base_pose_ground_truth" to="$(arg robot_name)/odom"/>
    </node>

   <node pkg="map_server" type="map_server" name="map_server"
          args="$(find ariac_environment)/maps/ariac.yaml">
      <param name="frame_id" value="$(arg robot_name)/map"/>
    </node>

  <include file="$(find guide_control)/launch/rviz.launch">
    <arg name="config" value="localization"/>
  </include>

  <include file="$(find guide_control)/launch/upload_actor.xml">
    <arg name="actor_name" value="$(arg actor_name)"/>
    <arg name="start_x" value="8"/>
    <arg name="start_y" value="9.6"/>
  </include>

  <node name="guide_controller" pkg="guide_control" type="guide_control.py" output="screen">
    <param name="~guide_distance" type="double" value="2"/>
    <param name="~actor_tolerance" type="double" value="0.2"/>
    <param name="~map_frame" value="$(arg robot_name)/map"/>
    <param name="~robot_frame" value="$(arg robot_name)/base_link"/>
    <remap from="odom" to="$(arg robot_name)/odom"/>
    <remap from="cmd_vel" to="$(arg robot_name)/cmd_vel"/>
  </node>

  <include file="$(find timed_roslaunch)/launch/timed_roslaunch.launch">
    <arg name="time" value="20"/>
    <arg name="pkg" value="guide_control"/>
    <arg name="file" value="detect_obstacles.launch"/>
  </include>

</launch>
